src/web/lib/hakuneko/frontend@classic-dark/mangas.html

```diff
@@ -262,11 +295,41 @@
                 //
             }
 
+            /**
+            * Event listener attached to the refresh button
+            * Checks for new content to read by refreshing the bookmarks, updating the web source and checking the chapters bookmarks
+            */
+            async onCheckNewContent() {
+                if(! this.get('enableCheckNewContent'))
+                {
+                    alert('Please wait 60s before checking again.');
+                    return;
+                }
+                this.set('nbContentUpdating',Engine.BookmarkManager.bookmarks.length); 
+                await Engine.BookmarkManager.checkNewContent();
+                this.set('nbContentUpdating',0);
+                this.set('enableCheckNewContent',false);
+                //Disable refresh for 1 minute to prevent hammering
+                setTimeout(function(context){
```

Comment:
I don't understand the 60 second timer.
The content check is awaitable, so after this is done the flag can just be set to true directly without using the timer?

Reply:
During my testing, i've found out that i'm tempted to click on the button to check again for new content just after checking it (yeah that's basically stupid). That would hammer the website when you have quite a long list of bookmarks (i have 200 bookmarks, and they get updated in less than 10s).
Disabling for 1min was the nicest option i've figured out (but it should also be disabled as long as the check content is running). The 1 min wait is not here to wait for the full refresh to end.